readthedocs/projects/validators.py

```diff
@@ -227,3 +230,19 @@ def _clean_prefix(prefix):
     if not prefix:
         return "/"
     return f"/{prefix}/"
+
+
+def validate_environment_variable_size(
+    project, new_env_value, error_class=ValidationError
+):
+    existing_size = sum(
+        project.environmentvariable_set.annotate(size=Length("value")).values_list(
+            "size", flat=True
+        )
+    )
```

Comment:
We could do `.aggregate(Sum("size"))` here if we want to do everything at db level.